lib/api/actions/functions.php

```diff
@@ -91,130 +91,122 @@ function beans_add_smart_action( $hook, $callback, $priority = 10, $args = 1 ) {
+ * {@see beans_add_smart_action()}. Each optional argument must be set to NULL to keep the original value.
  *
  * The original action can be reset using {@see beans_reset_action()}.
  *
  * @since 1.0.0
+ * @since 1.5.0 Made WPCS compliant and improved action parameter filtering.
+ *
+ * @param string        $id       The action's Beans ID, a unique ID tracked within Beans for this action.
+ * @param string|null   $hook     Optional. The new action's event name to which the $callback is hooked.
+ *                                Use NULL to keep the original value.
+ * @param callable|null $callback Optional. The new callback (function or method) you wish to be called.
+ *                                Use NULL to keep the original value.
+ * @param int|null      $priority Optional. The new priority.
+ *                                Use NULL to keep the original value.
+ * @param int|null      $args     Optional. The new number of arguments the $callback accepts.
+ *                                Use NULL to keep the original value.
  *
- * @param string   $id       The action ID.
- * @param string   $hook     Optional. The name of the new action to which the $callback is hooked.
- *                           Use NULL to keep the original value.
- * @param callback $callback Optional. The name of the new function you wish to be called.
- *                           Use NULL to keep the original value.
- * @param int      $priority Optional. The new priority.
- *                           Use NULL to keep the original value.
- * @param int      $args     Optional. The new number of arguments the function accept.
- *                           Use NULL to keep the original value.
- *
- * @return bool Will always return true.
+ * @return bool
  */
 function beans_modify_action( $id, $hook = null, $callback = null, $priority = null, $args = null ) {
-
-	// Remove action.
-	if ( $current = _beans_get_current_action( $id ) ) {
-		remove_action( $current['hook'], $current['callback'], $current['priority'], $current['args'] );
+	$action = _beans_build_action_for_valid_args( $hook, $callback, $priority, $args );
+	// If no changes were passed in, there's nothing to modify. Bail out.
+	if ( empty( $action ) ) {
+		return false;
 	}
 
-	$action = array_filter( array(
-		'hook'     => $hook,
-		'callback' => $callback,
-		'priority' => $priority,
-		'args'     => $args,
-	) );
-
-	// Merge modified.
-	$action = _beans_merge_action( $id, $action, 'modified' );
+	$current_action     = _beans_get_current_action( $id );
+	$has_current_action = ! empty( $current_action ) && is_array( $current_action );
 
-	// Replace if needed.
-	if ( $current ) {
-
-		$action = array_merge( $current, $action );
+	// If the action is registered, let's remove it.
+	if ( $has_current_action ) {
+		remove_action( $current_action['hook'], $current_action['callback'], $current_action['priority'], $current_action['args'] );
```

Comment:
Let's improve this by doing `call_user_func_array( 'remove_action', $current_action );`.

Reply:
While it looks like less code, directly invoking the function is a wee bit _faster_.  Why? PHP has to look up the function and then invoke it with `call_user_func_array`.

Reply:
That is strange, I got the exact same speed and memory usage when running quick tests before I commented on this. Did you setup some tests too? If yes, which PHP version are you running?

Reply:
That said, it might be that I got the same perf results because I only tested with one removal. Performance might be affect when it scales so let's leave it as it is ;-)